q2_amrfinderplus/annotate.py

```diff
@@ -0,0 +1,129 @@
+from typing import Union
+
+from q2_types.feature_data_mag import MAGSequencesDirFmt
+from q2_types.genome_data import (
+    GenesDirectoryFormat,
+    LociDirectoryFormat,
+    ProteinsDirectoryFormat,
+)
+from q2_types.per_sample_sequences import ContigSequencesDirFmt, MultiMAGSequencesDirFmt
+
+from q2_amrfinderplus.types import (
+    AMRFinderPlusAnnotationsDirFmt,
+    AMRFinderPlusDatabaseDirFmt,
+)
+from q2_amrfinderplus.utils import (
+    _create_empty_files,
+    _create_sample_dict,
+    _create_sample_dirs,
+    _get_file_paths,
+    _run_amrfinderplus_analyse,
+    _validate_inputs,
+)
+
+
+def annotate(
+    amrfinderplus_db: AMRFinderPlusDatabaseDirFmt,
+    sequences: Union[
+        MultiMAGSequencesDirFmt, ContigSequencesDirFmt, MAGSequencesDirFmt
+    ] = None,
+    proteins: ProteinsDirectoryFormat = None,
+    loci: LociDirectoryFormat = None,
+    organism: str = None,
+    plus: bool = False,
+    report_all_equal: bool = False,
+    ident_min: float = None,
+    curated_ident: bool = False,
+    coverage_min: float = 0.5,
+    translation_table: str = "11",
```

Comment:
Yes it is always an integer. But with integers you can only use the function Range() where you can specify a range of integers with start and end. But because there are numbers missing from 17 to 20 a range from 1 to 33 is not appropriate. A workaround is to specify it as strings and with that you can use Choices(). This does not change anything for the user as in the cli you can still specify translation_table without "" and it still works.

Reply:
Alright, makes sense!